tools/c7n_gcp/c7n_gcp/mu.py

```diff
@@ -369,14 +373,18 @@ def __init__(self, session, data=None):
     def prefix(self):
         return self.data.get('prefix', 'custodian-auto-')
 
-    def add(self, func):
+    def before(self, func):
```

Comment:
this name change feels a bit arbitrary, and non semantically correct. it aligns to call site interceptors, but its non universal ie. before/after apply to both create/delete/remove etc, but thats not what this pr is doing with those methods. the previous was goal state provisioning transition actions ie add (effectively ensure up to date), remove, which is also mirrored by other c7n providers doing resource provisioning (aws lambda) and powers common cleanup capabilities.

Reply:
yeah that is a good point. i was playing around with the before and after because in my first try at the pr i did need some provisioning after, but ended up not needing it. will revert those back to what it was before.